### PR TITLE
Add lecture api

### PR DESCRIPTION
**Pull request by shakti97**
Added the lecture api

### CODE REVIEW

File: routes/lecture.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+const { Lecture, validate } = require("../models/Lecture")
+const router = require("express").Router()
+const teacherAuth = require("../middlewares/teacherAuth")
+const auth = require("../middlewares/auth")
+const jwt = require("jsonwebtoken")
+const _ = require("lodash")
+
+router.get(
+  "/",
+  () => auth(req, res, next, false),
```
Comment (siddhant1) [inline routes/lecture.js]: the middleware will get req,res,next , pass that to custom middleware

File: routes/lecture.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      return
+    }
+    const lecture = await Lecture.find({ isPublished: true })
+    res.json(lecture)
+  }
+)
+
+router.get(
+  "/:id",
+  () => auth(req, res, next, false),
```
Comment (siddhant1) [inline routes/lecture.js]: same as above

Comment (shakti97) [reply routes/lecture.js]: oaky

File: routes/lecture.js (hunk 1 above)
Comment (shakti97) [reply routes/lecture.js]: okay

Comment (siddhant1) [inline routes/lecture.js]:
```suggestion
  (req,res,next) => auth(req, res, next, false),
```

File: routes/lecture.js (hunk 2 above)
Comment (siddhant1) [inline routes/lecture.js]:
```suggestion
  (req,res,next) => auth(req, res, next, false),
```

**Pull request merged**